File: 2_semestre/IHC/Pratica/IHC-Proj/AREC/app/src/main/java/com/example/arec/model/Event.kt
package com.example.arec.model

import com.google.android.gms.maps.model.LatLng

class Event {
    var eventName: String = ""
    var ID: String = ""                    //ID of the event
    var eventOwner: String = ""            //ID of the user
    var radius: Double = 0.0             //radius of the Event
    var latitude : Double = 0.0
    var longitude: Double = 0.0
    var duration: Double = 0.0           //duration of the event
    var totalParticipants: Int = 0
    val participants = mutableListOf<String>() //Participants on event real Time

    constructor() {
        // No-argument constructor for Firebase deserialization
    }

    constructor(
        eventName: String,
        ID: String,
        eventOwner: String,
        radius: Double,
        latlng: LatLng,
        duration: Double,
        totalParticipants: Int
    ) {
        this.eventName = eventName
        this.ID = ID
        this.eventOwner = eventOwner
        this.radius = radius
        this.latitude = latlng.latitude
        this.longitude = latlng.longitude
        this.duration = duration
        this.totalParticipants = totalParticipants
    }
    //toString
    override fun toString(): String {
        return "Event(ID=$ID, eventOwner=$eventOwner, radius=$radius, " +
                ", lat=$latitude, long = $longitude,  duration=$duration, " +
                "totalParticipants=$totalParticipants, " +
                "participants=$participants)\n"
    }


}

File: 2_semestre/ICM/spark-flutter/spark/lib/models/Event.dart
class Event {
  late String eventName;
  late String ID;
  late String eventOwner;
  late String eventDescription;
  String photoURL = "";
  late int radius;
  late double latitude;
  late double longitude;
  late int totalParticipants;
  late List<String> participants;

  Event({
    required this.eventName,
    required this.ID,
    required this.eventOwner,
    required this.eventDescription,
    required this.radius,
    required this.latitude,
    required this.longitude,
    required this.totalParticipants,
    this.participants = const [],
  });

  Map<String, dynamic> toMap() {
    return {
      'eventName': eventName,
      'ID': ID,
      'eventOwner': eventOwner,
      'eventDescription': eventDescription,
      'radius': radius,
      'latitude': latitude,
      'longitude': longitude,
      'totalParticipants': totalParticipants,
      'participants': participants,
      'photoURL': photoURL,
    };
  }

  Event.fromMap(Map<dynamic, dynamic> map) {
    eventName = map['eventName'];
    ID = map['ID'];
    eventOwner = map['eventOwner'];
    eventDescription = map['eventDescription'];
    radius = map['radius'];
    latitude = map['latitude'].toDouble();
    longitude = map['longitude'].toDouble();
    totalParticipants = map['totalParticipants'];
    photoURL = map['photoURL'];
    participants = List<String>.from(map['participants'] ?? []);
  }
}
